src/ape/utils/os.py

```diff
@@ -139,3 +139,14 @@ def __exit__(self, *exc):
         for path in self.exclude:
             if path not in sys.path:
                 sys.path.append(path)
+
+
+def get_full_extension(path: Path) -> str:
```

Comment:
Think if `path` is a directory, can just bail early w/ `""`

Reply:
We probably can. However, just as a note: ChainLink contains directories with `.sol` extensions for whatever reason.